src/components/stake/StakeValidatorListItem.vue

```diff
@@ -0,0 +1,228 @@
+        <div class="validator-item-wrapper">
+            <div class="validator-left">
+                <component v-bind:is="Icons[Helpers.capitalise(validatorData.icon) + 'Icon']" class="tab"></component>
+            </div>
+            <div class="validator-item-mid">
+                <div class="validator-item-inner-row validator-label">
+                    {{ validatorData.label }}
+                </div>
+                <div class="validator-item-inner-row validator-trust">
+                    <div :class="(validatorData.trust < 2.5)?'red-star':'silver-star'" class="star">
+                        <StarIcon />
+                    </div>
+                    <div :class="(validatorData.trust > 2.5)?'gold-text':'red-text'">
+                        {{ validatorData.trust.toFixed(2) }}
+                    </div>
+                    <div class="validator-payout">
+                        {{ payoutText }}
+                    </div>
+                </div>
+            </div>
+            <div :class="validatorData.reward < 2.5?'validator-warning':''" class="validator-item-right">
+                {{ validatorData.reward.toFixed(1) }} % {{ $t("p.a.") }}
+            </div>
+        </div>
+    </button>
+</template>
+
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api';
+import { numberToLiteralTimes } from '../../lib/NumberFormatting';
+
+/* mock data icons */
+import OverstakeIcon from '../icons/Staking/OverstakeIcon.vue';
+import NimiqWatchIcon from '../icons/Staking/NimiqWatchIcon.vue';
+import NimpoolIcon from '../icons/Staking/NimpoolIcon.vue';
+import AceStakingIcon from '../icons/Staking/AceStakingIcon.vue';
+import IceStakingIcon from '../icons/Staking/IceStakingIcon.vue';
+/* mock data icons */
+
+import StarIcon from '../icons/Staking/StarIcon.vue';
+
+const getPayoutText = (payout: Array<number>) => {
+    const periods = ['year', 'month', 'week', 'day', 'h'];
+    let index = 0;
+    let value = 0;
+
+    for (let i = 0; i < payout.length; i++) {
+        if (payout[i] > 0) {
+            index = i;
+            value = payout[i];
+            break;
+        }
+    }
+
+    if (index === periods.length - 1) {
+        return `pays out every ${value}${periods[index]}`;
+    }
+    return `pays out ${numberToLiteralTimes(value)} a ${periods[index]}`;
+};
+
+export default defineComponent({
+    setup(props) {
+        return {
+            payoutText: getPayoutText(props.validatorData.payout),
+            Helpers: {
+                capitalise: (string: string) => string.charAt(0).toUpperCase() + string.slice(1),
+            },
+            Icons: {
+                OverstakeIcon,
+                NimiqWatchIcon,
+                NimpoolIcon,
+                AceStakingIcon,
+                IceStakingIcon,
+            },
+        };
+    },
+    props: {
+        validatorData: {
+            type: Object,
+            required: true,
+        },
```

Comment:
Might want to make the properties of `validatorData` separate props to be more specific about which properties are expected.
Or could add a [prop validation method](https://vuejs.org/v2/guide/components-props.html#Prop-Validation) that checks for the expected properties.